src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -152,44 +87,147 @@ function calculatePrice(_numerator?: string | null, _denominator?: string | null
+}
+
+const UnfilledAmount: React.FC<UnfilledAmountProps> = ({ sellToken, order, pending }) => {
   const unfilledAmount = useMemo(() => formatAmount(order.remainingAmount, sellToken.decimals) || '0', [
     order.remainingAmount,
     sellToken.decimals,
   ])
+  const unlimited = useMemo(() => order.priceDenominator.gt(MIN_UNLIMITED_SELL_ORDER), [order.priceDenominator])
+
+  return (
+    <div className={'container' + (unlimited ? '' : ' sub-columns two-columns')}>
+      {unlimited ? (
+        <Highlight color={pending ? 'grey' : ''}>no limit</Highlight>
+      ) : (
+        <>
+          <div>{unfilledAmount}</div>
+          <div>
+            <strong>{displayTokenSymbolOrLink(sellToken)}</strong>
+          </div>
+        </>
+      )}
+    </div>
+  )
+}
+
+interface AccountBalanceProps extends Pick<Props, 'order' | 'isOverBalance'> {
+  sellToken: TokenDetails
+}
+
+const AccountBalance: React.FC<AccountBalanceProps> = ({ sellToken, order, isOverBalance }) => {
   const accountBalance = useMemo(() => formatAmount(order.sellTokenBalance, sellToken.decimals) || '0', [
     order.sellTokenBalance,
     sellToken.decimals,
   ])
-  const unlimitedAmount = useMemo(() => order.priceDenominator.gt(MIN_UNLIMITED_SELL_ORDER), [order.priceDenominator])
 
+  return (
+    <div className="container sub-columns three-columns">
+      <div>{accountBalance}</div>
+      <strong>{displayTokenSymbolOrLink(sellToken)}</strong>
+      <div className="warning">{isOverBalance && <FontAwesomeIcon icon={faExclamationTriangle} />}</div>
+    </div>
+  )
+}
+
+const Expires: React.FC<Pick<Props, 'order' | 'pending'>> = ({ order, pending }) => {
   const { isNeverExpires, expiresOn } = useMemo(() => {
     const isNeverExpires = isBatchIdFarInTheFuture(order.validUntil)
     const expiresOn = isNeverExpires ? '' : formatDateFromBatchId(order.validUntil)
+
     return { isNeverExpires, expiresOn }
   }, [order.validUntil])
 
+  return <div>{isNeverExpires ? <Highlight color={pending ? 'grey' : ''}>Never</Highlight> : expiresOn}</div>
+}
+
+async function fetchToken(
+  tokenId: number,
+  networkId: number,
+  setFn: React.Dispatch<React.SetStateAction<TokenDetails | null>>,
+): Promise<void> {
+  const token = await getTokenFromExchangeById({ tokenId, networkId })
+
+  // It is unlikely the token ID coming form the order won't exist
+  // Still, if that ever happens, store null and keep this order hidden
+  setFn(token)
+}
+
+interface Props {
+  order: AuctionElement
+  isOverBalance: boolean
+  networkId: number
+  pending?: boolean
+}
+
+const OrderRow: React.FC<Props> = props => {
+  const { order, networkId, pending = false } = props
+
+  // Fetching buy and sell tokens
+  const [sellToken, setSellToken] = useState<TokenDetails | null>(null)
+  const [buyToken, setBuyToken] = useState<TokenDetails | null>(null)
+
+  useEffect(() => {
+    fetchToken(order.buyTokenId, networkId, setBuyToken)
+    fetchToken(order.sellTokenId, networkId, setSellToken)
```

Comment:
These two fetch tokens are async, if there's an error, how is it handled?

Reply:
The order is only displayed when both tokens are known.
If there's a failure the order will remain hidden.

Reply:
Mmm, I'm not sure about this. Everytime there's a promise that fetches info, there can be an exception, so it this exception happens, the user won't see any input, right?

Reply:
Right. In this case, I can show a pop up notification on error.
Should we also notify in case the address referenced in the order is not a token or some other kind of edge case like this?

Reply:
Although it's weird I would say that this exception should trigger an alert modal or something. We hopefully won't see ever those ever. But if they show, we can review if we can handle better the case.

One issue is that it's likely that we see some repeated patterns and we will need in many places to show an error for weird cases, so maybe we can do something more or less general (like a general error handler we can give to all those promises).

Reply:
Created a generic utils error handler.

Also warning when (in the unlikely event) a token is not ERC20 compliant:

![Screenshot_20191219_105907](https://user-images.githubusercontent.com/43217/71201654-54e0c980-224f-11ea-86f8-5beebe114bab.png)